elyra/pipeline/processor_local.py

```diff
@@ -0,0 +1,130 @@
+    """
+    _type = 'local'
+
+    def __init__(self, root_dir=None):
+        super().__init__()
+        self._root_dir = root_dir
+
+    @property
+    def type(self):
+        return self._type
+
+    def process(self, pipeline):
+        timestamp = datetime.now().strftime('%m%d%H%M%S')
+        pipeline_name = f'{pipeline.name}-{timestamp}'
+
+        pipeline_work_dir = LocalPipelineProcessor._get_pipeline_work_dir(pipeline_name)
+        self.log.info(f'Processing Pipeline : {pipeline.name} at {pipeline_work_dir}')
+
+        operations = LocalPipelineProcessor._get_operations_by_dependency(pipeline.operations)
+        for operation in operations:
+            notebook = self.get_absolute_path(operation.filename)
+            if not os.path.isfile(notebook):
+                raise FileNotFoundError(f'Could not find {notebook}')
+
+            self._execute_file(pipeline_work_dir, notebook)
+
+        return PipelineProcessorResponse('', '', object_storage_path=pipeline_work_dir)
+
+    def export(self, pipeline, pipeline_export_format, pipeline_export_path, overwrite):
+        raise NotImplementedError('Local pipelines does not support export functionality')
+
+    def _execute_file(self, work_dir, filename):
+        notebook_dir = os.path.dirname(filename)
+        notebook_name = os.path.basename(filename)
+        stdout = os.path.join(work_dir, f'{notebook_name}.out')
+        stderr = os.path.join(work_dir, f'{notebook_name}.err')
+
+        self.log.debug(f'Processing: {filename}')
+        with open(stdout, 'w') as stdout_file, open(stderr, 'w') as stderr_file:
+            t0 = time.time()
+            try:
+                # Try executing with configured python kernel
+                papermill.execute_notebook(
+                    input_path=filename,
+                    output_path=filename,
+                    stdout_file=stdout_file,
+                    stderr_file=stderr_file,
+                    cwd=notebook_dir
+                )
+            except KeyError:
+                # force default python kernel
+                papermill.execute_notebook(
+                    input_path=filename,
+                    output_path=filename,
+                    stdout_file=stdout_file,
+                    stderr_file=stderr_file,
+                    cwd=notebook_dir,
+                    kernel_name="python3"
+                )
+            except Exception:
+                self.log.error(f'Internal error executing {filename}')
+
+            t1 = time.time()
+            duration = (t1 - t0)
+            self.log.debug(f'Execution of {notebook_name} took {duration:.3f} secs.')
+
+    @staticmethod
+    def _get_pipeline_work_dir(pipeline_name: str) -> str:
+        log_dir = os.path.join(tempfile.tempdir, 'elyra', pipeline_name)
+        os.makedirs(log_dir, mode=0o700, exist_ok=True)
+        return log_dir
+
+    @staticmethod
+    def _get_operations_by_dependency(operations_by_id: dict) -> list:
```

Comment:
This method name is not intuitive.  Seems like `_sort_operations` or `_get_sorted_operations` is more straightforward (and intuitive).

Please add help string.